### PR TITLE
Enable Orbital to fix undefined payment

### PR DESCRIPTION
**Pull request by daliwei**
cc @pierre for review.

Also need to update the gem dependency on killbill-plugin-framework-ruby after the follow PR is reviewed.

Together with https://github.com/killbill/killbill-plugin-framework-ruby/pull/63

### CODE REVIEW

File: db/ddl.sql [hunk 1]
```diff
@@ -117,6 +117,7 @@ CREATE TABLE orbital_responses (
   params_terminal_id varchar(255) DEFAULT NULL,
   params_tx_ref_idx varchar(255) DEFAULT NULL,
   params_tx_ref_num varchar(255) DEFAULT NULL,
+  params_trace_number varchar(16) DEFAULT NULL,
```
Comment (pierre) [inline db/ddl.sql:120]: Is that always guaranteed to be less than 16 chars? If not, might as well set it to 255 (storage is cheap).

File: lib/orbital/api.rb [hunk 2]
```diff
@@ -52,39 +54,39 @@ def purchase_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb
...
 
         properties = merge_properties(properties, options)
-        super(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_payment_method_id, amount, currency, properties, context)
+        super(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_payment_method_id, amount, currency, properties, context, with_trace_num_and_order_id(properties))
       end
 
       def get_payment_info(kb_account_id, kb_payment_id, properties, context)
-        # Pass extra parameters for the gateway here
-        options = {}
+        options = properties_to_hash properties
```
Comment (pierre) [inline lib/orbital/api.rb]: Here and below: could add parens for code clarity?

File: lib/orbital/api.rb [hunk 3]
```diff
@@ -142,6 +144,82 @@ def search_payment_methods(search_key, offset, limit, properties, context)
...
+        orbital_order_id = plugin_trx_info.second_payment_reference_id
+
+        return nil if trace_number.nil? || orbital_order_id.nil? || payment_processor_account_id.nil?
+
+        gateway = lookup_gateway(payment_processor_account_id, context.tenant_id)
+        gateway.inquiry(orbital_order_id, trace_number)
+      end
+
+      def with_trace_num_and_order_id(properties)
+        {:params_trace_number => find_value_from_properties(properties, :trace_number),
```
Comment (pierre) [inline lib/orbital/api.rb:219]: It looks like Orbital requires two `ids` to retrieve payment information? What is the difference between the two?

File: lib/orbital/api.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 module Killbill #:nodoc:
   module Orbital #:nodoc:
     class PaymentPlugin < ::Killbill::Plugin::ActiveMerchant::PaymentPlugin
+      FIVE_MINUTES_AGO = (1 * 300)
```
Comment (pierre) [inline lib/orbital/api.rb]: Nit: `1 * 300` a bit odd (vs `5 * 60`)

File: db/ddl.sql (hunk 1 above)
Comment (daliwei) [reply db/ddl.sql:120]: Yes, it is a number between 1-9999999999999999

File: lib/orbital/api.rb (hunk 3 above)
Comment (daliwei) [reply lib/orbital/api.rb:219]: One Order Id can have multiple trace numbers that correspond to each request we send.

File: lib/orbital/api.rb (hunk 4 above)
Comment (daliwei) [reply lib/orbital/api.rb]: Right. Fix in bcfd57a

File: lib/orbital/api.rb (hunk 2 above)
Comment (daliwei) [reply lib/orbital/api.rb]: Done in ac027eb

**Review by pierre — APPROVED** [review]

**Pull request merged**